truffleruby/src/main/ruby/core/file.rb

```diff
@@ -1348,9 +1348,30 @@ def world_writable?
 end
 
 STDIN = File.new(0)
+
+def STDIN.read(length, outbuf=nil)
+  bytes = Truffle::Interop.polyglot_in_read length
+  return nil unless bytes
+  if outbuf
+    outbuf << bytes
+    outbuf
+  else
+    bytes
+  end
+end
+
 STDOUT = File.new(1)
+
+def STDOUT.write(bytes)
+  Truffle::Interop.polyglot_out_write bytes
+end
+
 STDERR = File.new(2)
 
+def STDERR.write(bytes)
+  Truffle::Interop.polyglot_err_write bytes
+end
```

Comment:
This will not work nicely if sby `IO#reopen` one of the standard streams.
I think this should be handled in the primitive read/write, and special case if `fd` < 3.